tasks/test_results_processor.py

```diff
@@ -0,0 +1,200 @@
+import base64
+import json
+import logging
+import random
+import zlib
+from copy import deepcopy
+from datetime import datetime
+from io import BytesIO
+from json import loads
+from typing import List
+
+from redis.exceptions import LockError
+from shared.celery_config import test_results_processor_task_name
+from shared.config import get_config
+from shared.yaml import UserYaml
+from sqlalchemy.orm import Session
+from testing_result_parsers import (
+    Testrun,
+    parse_junit_xml,
+    parse_pytest_reportlog,
+    parse_vitest_json,
+)
+
+from app import celery_app
+from database.models import Repository, Upload
+from database.models.reports import Test, TestInstance
+from services.archive import ArchiveService
+from services.yaml import read_yaml_field
+from tasks.base import BaseCodecovTask
+
+log = logging.getLogger(__name__)
+
+
+class TestResultsProcessorTask(BaseCodecovTask, name=test_results_processor_task_name):
+    __test__ = False
+
+    async def run_async(
+        self,
+        db_session,
+        *,
+        repoid,
+        commitid,
+        commit_yaml,
+        arguments_list,
+        report_code=None,
+        **kwargs,
+    ):
+        commit_yaml = UserYaml(commit_yaml)
+
+        repoid = int(repoid)
+        log.info(
+            "Received upload test result processing task",
+            extra=dict(repoid=repoid, commit=commitid),
+        )
+
+        testrun_list = []
+        upload_list = []
+
+        # process each report session's test information
+        for args in arguments_list:
+            upload_obj: Upload = (
+                db_session.query(Upload).filter_by(id_=args.get("upload_pk")).first()
+            )
+
+            try:
+                parsed_testruns: List[Testrun] = self.process_individual_arg(
+                    upload_obj, upload_obj.report.commit.repository
+                )
+            except Exception:
```

Comment:
It's fine to leave a catch all exception for starters so we avoid crashing the task while we battle test it. But please be explicit about it, as having a catch-all exception is not good practice.

Also the warning message won't be helpful when debugging. Make sure to include the stack trace so we know what to fix (in case there's anything)